src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java

```diff
@@ -679,6 +680,41 @@ public void testExternalResourceFiltersWithNoExternalResource() throws Exception
         verify(checkerConfig, filePath, expected);
     }
 
+    /**
+     * This test is required for pitest coverage.
+     * Order of Check execution is not visible from outside
+     * as we order violations by line and column.
+     * This test is checking that Checks execution ordered by name and then by id.
+     *
+     * @throws Exception if file is not found
+     */
+    @Test
+    public void testOrderOfCheckExecution() throws Exception {
+
+        final DefaultConfiguration configuration1 = createModuleConfig(AaCheck.class);
+        configuration1.addProperty("id", "2");
+        final DefaultConfiguration configuration2 = createModuleConfig(BbCheck.class);
+        configuration2.addProperty("id", "1");
+
+        final DefaultConfiguration treeWalkerConfig = createModuleConfig(TreeWalker.class);
+        treeWalkerConfig.addChild(configuration2);
+        treeWalkerConfig.addChild(configuration1);
+
+        final List<File> files =
+                Collections.singletonList(new File(getPath("InputTreeWalker2.java")));
+        final Checker checker = createChecker(treeWalkerConfig);
+
+        try {
+            checker.process(files);
```

Comment:
Why can't this be a BDD test? I am not seeing anything really preventing it.

Reply:
All tries to make it in bad style did not kill mutation. Because mutation on order of checks, order of checks is not publicly visible.
I tried bunch and gave up.

Reply:
Have we tried something like the following?
- config with two checks
- both checks have bad config (maybe property name)
- we assert that it is the first check in the config that reported in the exception
?

Reply:
I am not , but exception will happen before Check is added to sorted-set of Checks of Treewker.
Order of check to load from config is defined by xml, predictable order, but mutation is for order of execution in runtime (traverse of tree).

I tried to keep test clear that first execution on beginTree will throw exception, so it very clear definition that Checks are ordered.
After bunch of time with this mutation, I am very ok to keep it as non bdd, because final variant of test is still easy to understand, I hope.

Reply:
Yes, test is easy to understand.